posthog/temporal/batch_exports/postgres_batch_export.py

```diff
@@ -35,120 +35,309 @@
     get_bytes_exported_metric,
     get_rows_exported_metric,
 )
-from posthog.temporal.batch_exports.temporary_file import (
-    BatchExportTemporaryFile,
+from posthog.temporal.batch_exports.temporary_file import CSVBatchExportWriter
+from posthog.temporal.batch_exports.utils import (
+    JsonType,
+    apeek_first_and_rewind,
+    cast_record_batch_json_columns,
+    set_status_to_running_task,
 )
-from posthog.temporal.batch_exports.utils import apeek_first_and_rewind, set_status_to_running_task
 from posthog.temporal.common.clickhouse import get_client
 from posthog.temporal.common.heartbeat import Heartbeater
 from posthog.temporal.common.logger import bind_temporal_worker_logger
 
+PostgreSQLField = tuple[str, typing.LiteralString]
```

Comment:
Yay Python 3.11, we now have `typing.LiteralString` and can get rid of a type: ignore comment.